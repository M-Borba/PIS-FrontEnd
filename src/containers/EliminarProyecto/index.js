import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { useSnackbar } from "notistack";

import DeleteDialogContent from "../../components/DeleteDialogContent";
import { axiosInstance } from "../../config/axios";
import { BUTTON_LABELS } from "../../config/globalVariables";

EliminarProyecto.propTypes = {
  projectName: PropTypes.string.isRequired,
  projectId: PropTypes.number.isRequired,
  handleClose: PropTypes.func.isRequired,
  removeRow: PropTypes.func.isRequired,
};

function EliminarProyecto({ projectId, projectName, handleClose, removeRow }) {
  const dialogContent = `¿Está seguro que desea eliminar el proyecto ${projectName} del sistema?`;
  const { enqueueSnackbar } = useSnackbar();

  const onConfirmation = () => {
    axiosInstance
      .delete(`/projects/${projectId}`)
      .then(() => {
        removeRow(projectId);
        enqueueSnackbar(`El proyecto ${projectName} se eliminó con éxito.`, {
          variant: "success",
          autoHideDuration: 4000,
        });
        handleClose();
      })
      .catch((error) => {
        console.error(error.response);
        enqueueSnackbar(error.response.data?.error, {
          variant: "error",
          autoHideDuration: 8000,
        });
      });
  };

  return (
    <Fragment>
      <DeleteDialogContent
        dialogContent={dialogContent}
        onClose={handleClose}
        onConfirmation={onConfirmation}
        deleteButtonText={BUTTON_LABELS.DELETE}
      />
    </Fragment>
  );
}

export default EliminarProyecto;
