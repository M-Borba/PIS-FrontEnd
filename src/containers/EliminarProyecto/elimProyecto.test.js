import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ReactDOM from "react-dom";
import EliminarProyecto from "./index";
import React from "react";

describe("project list component tests", () => {
  it("renders without crashing", () => {
    const handleClose = () => {};
    const div = document.createElement("div");
    ReactDOM.render(
      <EliminarProyecto
        projectId={1}
        projectName={"Prueba"}
        handleClose={handleClose}
      />,
      div
    );
  });
  it("renders correct data", () => {
    const handleClose = () => {};
    const { container } = render(
      <EliminarProyecto
        projectId={1}
        projectName={"Prueba"}
        handleClose={handleClose}
      />
    );
    // Matches the last snapshot taken
    expect(container).toMatchSnapshot();
  });
});
