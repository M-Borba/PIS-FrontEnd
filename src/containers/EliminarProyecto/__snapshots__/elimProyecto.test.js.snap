// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project list component tests renders correct data 1`] = `
<div>
  <div
    class="MuiDialogTitle-root makeStyles-title-1"
    id="confirmation-dialog-title"
  >
    <h2
      class="MuiTypography-root MuiTypography-h6"
    >
      Confirmar eliminacion
      <button
        aria-label="Close"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-closeButton-4"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </h2>
  </div>
  <div
    class="MuiDialogContent-root makeStyles-content-2"
  >
    Esta seguro que desea eliminar el proyecto Prueba del sistema?
  </div>
  <div
    class="MuiDialogActions-root makeStyles-actions-3 MuiDialogActions-spacing"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Cancelar
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Confirmar
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
